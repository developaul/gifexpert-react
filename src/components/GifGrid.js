import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem, Spinner } from './';

const GifGrid = ({ nameCategory }) => {

    const { data:images, loading } = useFetchGifs( nameCategory );

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{ nameCategory }</h3>

            { loading && <Spinner /> }

            <div className="card-grid">
                {
                    images.map( img => (
                        <GifGridItem
                            key={ img.id }
                            nameCategory={ nameCategory }
                            { ...img }
                        />
                    ))
                }
            </div>
        </>
    );
};

GifGrid.propTypes = {
    nameCategory: PropTypes.string.isRequired
};

export default GifGrid;