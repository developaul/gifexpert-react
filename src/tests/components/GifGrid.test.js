import React from 'react';

import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock( '../../hooks/useFetchGifs' );

describe( 'Pruebas en el componente <GifGrid />', () => {

    test( 'Debe de renderizar correctamente el componente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true 
        });
            
        const wrapper = shallow( <GifGrid nameCategory='One Punch' /> );    
        expect( wrapper ).toMatchSnapshot();

    });
    
    test( 'Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/testing.jpg',
            title: 'testing.jpg'
        },
        {
            id: 'ABCDE',
            url: 'https://localhost/testing.jpg',
            title: 'testing.jpg'
        },];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
            
        const wrapper = shallow( <GifGrid nameCategory='One Punch' /> );    

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'Spinner' ).exists() ).toBe( false );
        expect( wrapper.find( 'GifGridItem' ).length ).toBe( gifs.length );
    });

});