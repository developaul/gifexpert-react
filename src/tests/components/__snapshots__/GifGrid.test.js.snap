// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      nameCategory="One Punch"
      title="testing.jpg"
      url="https://localhost/testing.jpg"
    />
    <GifGridItem
      id="ABCDE"
      key="ABCDE"
      nameCategory="One Punch"
      title="testing.jpg"
      url="https://localhost/testing.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> Debe de renderizar correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <Spinner />
  <div
    className="card-grid"
  />
</Fragment>
`;
